vm/src/vm.rs

```diff
@@ -326,19 +331,32 @@ impl GlobalVMState {
         vm
     }
 
-    fn add_types(&self) -> StdResult<(), (TypeId, TcType)> {
+    fn add_types(&mut self) -> StdResult<(), (TypeId, TcType)> {
         use api::generic::A;
         use api::Generic;
+        fn add_type<T: Any>(ids: &mut HashMap<TypeId, TcType>,
+                            env: &mut VMEnv,
+                            name: &str,
+                            typ: TcType) {
+            ids.insert(TypeId::of::<T>(), typ);
+            // Insert aliases so that `find_info` can retrieve information about the primitives
+            env.type_infos.id_to_type.insert(name.into(),
+                                             types::Alias::from(types::AliasData {
```

Comment:
Might look nicer if you imported these things from `types` in the top of the scope.